dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java

```diff
@@ -297,33 +299,58 @@ protected boolean rewriteDockerfile(String img, String tag,
      * {@code stringBuilder} will get {@code line} added to it. We return {@code false} in this instance.
      *
+     *
+     * @param reader BufferReader used for reading next line to FROM Instruction
      * @param imageToFind the Docker image that may require a tag update
      * @param tag the Docker tag that we'd like the image to have
      * @param stringBuilder the stringBuilder to accumulate the output lines for the pull request
      * @param line the inbound line from the Dockerfile
      * @return Whether we've modified the {@code line} that goes into {@code stringBuilder}
      */
-    protected boolean changeIfDockerfileBaseImageLine(String imageToFind, String tag,
+    protected boolean changeIfDockerfileBaseImageLine(BufferedReader reader, String imageToFind, String tag,
                                                       StringBuilder stringBuilder, String line,
-                                                      String ignoreImageString) {
+                                                      String ignoreImageString) throws IOException {
         boolean modified = false;
         String outputLine = line;
+        String nextLine = null;
 
         // Only check/modify lines which contain a FROM instruction
         if (FromInstruction.isFromInstruction(line)) {
             FromInstruction fromInstruction = new FromInstruction(line);
+            nextLine = reader.readLine();
             if (fromInstruction.hasBaseImage(imageToFind) &&
                     fromInstruction.hasADifferentTag(tag) &&
-                    !fromInstruction.ignorePR(ignoreImageString)) {
+                    !ignorePR(nextLine, ignoreImageString)) {
                 fromInstruction = fromInstruction.getFromInstructionWithNewTag(tag);
                 modified = true;
             }
             outputLine = fromInstruction.toString();
         }
         stringBuilder.append(outputLine).append("\n");
+        if (nextLine != null)
+            stringBuilder.append(nextLine).append("\n");
         return modified;
     }
 
+    /**
+     * Determines whether a comment after FROM line is mentioned {@code ignoreImageString} to ignore creating dfiu PR
+     * If {@code ignoreImageString} present in comment, PR should be ignored
+     * If {@code ignoreImageString} is empty, then by default 'no-dfiu' comment will be searched
+     * @param nextLine Next line to search for comment
+     * @param ignoreImageString comment to search
+     * @return {@code true} if comment is found
+     */
+    protected boolean ignorePR(String nextLine, String ignoreImageString) throws IOException {
+        if (nextLine != null && nextLine.trim().startsWith("#")) {
+            if (StringUtils.isNotBlank(ignoreImageString)) {
+                return nextLine.contains(ignoreImageString);
+            } else {
+                return nextLine.contains(NO_DFIU);
+            }
+        }
+        return false;
+    }
```

Comment:
You can simplify this quite a bit.

```
final String tester = Optional.ofNullable(ignoreImageString).orElse(NO_DFIU);
return StringUtils.isNotBlank(nextLine) && nextLine.contains(tester)
```

Reply:
Done as suggested.
And yes I think we should not restrict to exact comment line, rather any comment line containing the exact ignore string matching below should be OK.
#no-dfiu added to ignore PR
# Adding "no-dfiu" to ignore PR
Thoughts?